keps/sig-auth/2799-reduction-of-secret-based-service-account-token/README.md

```diff
@@ -0,0 +1,465 @@
+
+## Motivation
+
+As BoundServiceAccountTokenVolume is GA in 1.22, pods’ service account tokens
+would be obtained via TokenRequest API and stored as projected volume. This
+change obviates the need for auto-generation of secret-based service account
+tokens which are [less secure than the bound token](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/1205-bound-service-account-tokens#background).
+
+### Goals
+
+- No auto-generation of secret-based service account token.
+- Removal of unused auto-generated secret-based service account tokens
+
+### Non-Goals
+
+- Removal of [explicitly requested secret-based service account tokens](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-a-service-account-api-token).
+
+## Proposal
+
+- Change the service account control loop in Token Controller to not create
+  secret for service accounts. At the same time, warn usage of auto-created
+  secret-based service account tokens and encourage users to use TokenRequest
+  API or manually-created secret-based service account tokens.
+- Purge unused auto-generated secret-based service account tokens.
+
+### Notes/Constraints/Caveats
+
+- A warning mechanism should be implemented to help users
+  migrate.
+- Auto generated secret-based service account tokens are those requested by
+  Token Controller.
+- Only clean up auto-generated tokens which:
+  - are not referenced by pods
+  - have not been used to authenticate for some duration (time duration or number of releases)
+- To consult active usage of secret-based tokens, metric
+  `serviceaccount_legacy_tokens_total` or audit annotation
+  `authentication.k8s.io/legacy-token` could be used.
+
+### Risks and Mitigations
+
+- When feature LegacyServiceAccountTokenNoAutoGeneration is Beta, consumers
+  depending directly on waiting for and reading tokens out of auto-generated
+  secrets might stop working. To mitigate,
+  1. Emit warnings when using auto-generated token secrets.
+  2. Publish pointers to TokenRequest or the manual secret request flow.
+- When LegacyServiceAccountTokenCleanUp is Beta, usage of auto-generated
+  secret-based token might stop working. To mitigate,
+  1. When Alpha, annouce the cleanup starts at Beta
+  2. Emit warnings when using auto-generated token secrets.
+  3. Add pointers of TokenRequest API and manually created tokens in the validation
+     result.
+
+## Design Details
+
+### LegacyServiceAccountTokenNoAutoGeneration:
+
+Token Controller stops to create secret for service accounts. This feature would
+be enabled when it is implemented since no new code is added and this can make
+sure new clusters are in good state.
+
+### LegacyServiceAccountTokenTracking
+
+To facilitate LegacyServiceAccountTokenCleanUp,
+
+- When LegacyServiceAccountTokenTracking is disabled,
+
+  - create a configmap in `kube-system` namespace that stores the current
+    date as `tracked-since` if it doesn't exist yet.
```

Comment:
clarify it's the API server that does this (prior art for API server persisting data to a kube-system configmap in pkg/controlplane/controller/clusterauthenticationtrust/cluster_authentication_trust_controller.go)

also clarify when this happens (could either be done once at apiserver startup or periodically... I'd probably suggest periodically)